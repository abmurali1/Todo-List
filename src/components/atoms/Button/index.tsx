import React, {FC} from 'react';
import {SafeAreaView, StyleSheet, Text, TouchableOpacity} from 'react-native';

interface ButtonProps {
  title: string;
  onPress: () => void;
}

const Button: FC<ButtonProps> = props => {
  return (
    <TouchableOpacity style={styles.buttonStyle} onPress={props.onPress}>
      <Text allowFontScaling={false} style={styles.textStyle}>
        {props.title}
      </Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  buttonStyle: {
    backgroundColor: '#465c47',
    height: 50,
    width: 200,
    alignItems: 'center',
    justifyContent: 'center',
    flexDirection: 'row',
    borderRadius: 10,
  },
  textStyle: {
    color: '#ccb866',
    fontSize: 20,
    fontFamily: 'Verdana',
    letterSpacing: 3,
  },
});

export default Button;
