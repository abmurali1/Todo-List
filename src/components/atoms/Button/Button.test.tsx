import React from 'react';
import renderer from 'react-test-renderer';
import Button from './index';

describe('<Button>', () => {
  it('renders correctly', () => {
    const mockFn = jest.fn();
    const tree = renderer
      .create(<Button title="Click" onPress={mockFn} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
