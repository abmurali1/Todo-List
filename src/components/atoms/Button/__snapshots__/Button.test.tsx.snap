// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button> renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#465c47",
      "borderRadius": 10,
      "flexDirection": "row",
      "height": 50,
      "justifyContent": "center",
      "opacity": 1,
      "width": 200,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Object {
        "color": "#ccb866",
        "fontFamily": "Verdana",
        "fontSize": 20,
        "letterSpacing": 3,
      }
    }
  >
    Click
  </Text>
</View>
`;
